upstream-community-operators/robin-operator/robin.clusterserviceversion.yaml

```diff
@@ -0,0 +1,315 @@
+    mediatype: image/png
+  links:
+  - name: Robin
+    url: 'https://robin.io'
+  - name: User Guide
+    url: 'https://get.robin.io/openshift/robin_ug'
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      deployments:
+      - name: robin-operator
+        #namespace: default
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              name: robin-operator
+          template:
+            metadata:
+              labels:
+                name: robin-operator
+                app.kubernetes.io/name: robin
+            spec:
+              serviceAccountName: robin-operator
+              containers:
+                - name: robin-operator
+                  image: robinsys/robin-operator:5.1.0
+                  command:
+                  - robin-operator
+                  env:
+                    - name: WATCH_NAMESPACE
+                      valueFrom:
+                        fieldRef:
+                          fieldPath: metadata.annotations['olm.targetNamespaces']
+                    - name: POD_NAME
+                      valueFrom:
+                        fieldRef:
+                          fieldPath: metadata.name
+                    - name: OPERATOR_NAME
+                      value: robin-operator
+      clusterPermissions:
```

Comment:
Please reduce the permissions and clusterPermissions to the roles and clusterRoles that the operator needs to function. We will not accept operators that have all access to all APIs.
Also note that if in the clusterwide install the operator needs cluster permissions to view CRs in other namespaces, OLM automatically upgrades permissions to clusterPermissions when the operator is installed with the all namespaces install mode.

Reply:
Yes, we will work on this in parallel, will try to get this done by today. In the meantime is there anyway we can get the approval, we want to be live by the Redhat summit. Thanks

Reply:
Robin storage operator beyond providing storage for k8s applications deployed by helm charts supports following additional features:

- Snapshot
- Cloning
- Backup
- Import 

The above features include not only the storage volumes but all the k8s objects being used by an application deployed using helm chart. We can not know beforehand what k8s objects an application might be using, which might include custom resources. That's the reason we need permissions on all the objects.